pkg/extensions/extension_search.go

```diff
@@ -51,30 +62,105 @@ func EnableSearchExtension(config *config.Config, storeController storage.StoreC
 			log.Warn().Msg("CVE update interval set to too-short interval < 2h, changing update duration to 2 hours and continuing.") //nolint:lll // gofumpt conflicts with lll
 		}
 
-		go func() {
-			err := downloadTrivyDB(cveInfo, log, config.Extensions.Search.CVE.UpdateInterval)
-			if err != nil {
-				log.Error().Err(err).Msg("error while downloading TrivyDB")
-			}
-		}()
+		updateInterval := config.Extensions.Search.CVE.UpdateInterval
+
+		downloadTrivyDB(updateInterval, taskScheduler, cveInfo, log)
 	} else {
 		log.Info().Msg("CVE config not provided, skipping CVE update")
 	}
 }
 
-func downloadTrivyDB(cveInfo CveInfo, log log.Logger, updateInterval time.Duration) error {
-	for {
-		log.Info().Msg("updating the CVE database")
+func downloadTrivyDB(interval time.Duration, sch *scheduler.Scheduler, cveInfo CveInfo, log log.Logger) {
+	generator := &trivyTaskGenerator{interval, cveInfo, log, pending, 0, time.Now(), &sync.Mutex{}}
 
-		err := cveInfo.UpdateDB()
-		if err != nil {
-			return err
-		}
+	sch.SubmitGenerator(generator, interval, scheduler.HighPriority)
+}
+
+type trivyTaskGenerator struct {
+	interval     time.Duration
+	cveInfo      CveInfo
+	log          log.Logger
+	status       state
+	waitTime     time.Duration
+	lastTaskTime time.Time
+	lock         *sync.Mutex
+}
+
+func (gen *trivyTaskGenerator) SetTrivyGeneratorStatus(status state) {
```

Comment:
Not really sure we need the setter/getter anymore, since we don't lock here anymore.